code/modules/modular_computers/NTNet/device_types/_network_device.dm

```diff
@@ -0,0 +1,180 @@
+/datum/extension/network_device
+	base_type = /datum/extension/network_device
+	expected_type = /atom/movable
+	flags = EXTENSION_FLAG_IMMEDIATE
+	var/network_id
+	var/key
+	var/address
+	var/network_tag
+	var/connection_type = NETWORK_CONNECTION_WIRELESS
+
+/datum/extension/network_device/New(datum/holder, n_id, n_key, c_type, autojoin = TRUE)
+	..()
+	network_id = n_id
+	key = n_key
+	if(c_type)
+		connection_type = c_type
+	address = uppertext(NETWORK_MAC)
+	var/obj/O = holder
+	network_tag = "[uppertext(replacetext(O.name, " ", "_"))]-[num2hex(sequential_id(type))]"
```

Comment:
Not user friendly imo. In my branch I made default network tags procedural. Review that code to see what I meant.

net_tag wasn't ever assigned to because it was an "override" variable, if net_tag was null, no net_tag was assigned, so it always got the most up-to-date procedural tag which was always the device holder's name + the MAC address. So device name changes were reflected instantly.

Only if net_tag was assigned by a user (and thus overriding the procedural net_tag) would it display that, and even still the proc added the mac address onto the end, making sure it was completely and utterly unique no matter what the user set.

Reply:
I dunno if instantly checking network tag when machine name changes is desirable since I'm using it as 'address' for remote file browser, kinda weird that I can e.g. drop fileserver users by slapping it with a hand labeler.